packages/trader/src/Stores/Modules/Trading/trade-store.js

```diff
@@ -421,7 +423,19 @@ export default class TradeStore extends BaseStore {
                         type: response.msg_type,
                         ...response.error,
                     };
-                    this.root_store.ui.toggleServicesErrorModal(true);
+                    
+                    // check if token is invalid
+                    if (response.error.code === 'InvalidToken') {
+                    // send logout, clear contracts, clean up client and then show message
+                        this.root_store.client.cleanUp();
+                        requestLogout().then(() => {
+                            this.root_store.modules.trade.clearContracts();
```

Comment:
should cleanup happen after logout?

Reply:
I tried that but clean up removes error message